viper.go

```diff
@@ -1133,6 +1122,23 @@ func (v *Viper) find(lcaseKey string, skipDefault bool) interface{} {
 	return nil
 }
 
+// Given a key, find the value.
+// Viper will check in the following order:
+// flag, env, config file, key/value store, default.
+// If skipDefault is set to true, find will ignore default values.
+// Viper will check to see if an alias exists first.
+// Note: this assumes a lower-cased key given.
+func (v *Viper) find(lcaseKey string, skipDefault bool) interface{} {
+
+	if val := v.findExact(lcaseKey, skipDefault); val != nil {
+		return val
+	}
+
+	// if the requested key is an alias, then return the proper key
+	lcaseKey = v.realKey(lcaseKey)
+	return v.findExact(lcaseKey, skipDefault)
```

Comment:
```suggestion
	if val := v.findExact(lcaseKey, skipDefault); val != nil {
		return val
	}

	// if the requested key is an alias, then return the proper key
	realKey := v.realKey(lcaseKey)
	if realKey != lcaseKey {
	    return v.findExact(realKey, skipDefault)
	}
	return nil
```

`realKey` returns its parameter when no alias was found and `findExact` seems like a big function so we might as well not call it again.

Reply:
Not sure I understand your comment

simply calling
```go
v.findExact(v.realKey(lcaseKey), skipDefault)
```
is the old behavior, which didn't work for our use-case

Reply:
Ok I didn't notice that `findExact` already looked for aliases, but then I don't see how you could get a different return value by calling `v.findExact(lcaseKey, skipDefault)` and `v.findExact(v.realKey(lcaseKey), skipDefault)` ?

Reply:
at first, I'm looking for the exact value I got (the alias), if I didn't find it, I'm using `realKey` to resolve aliasing, and then looking again

Today's implementation will not look for the alias, but always to the resolved value

Reply:
so both of them can live together, the goal here is to allow "application backward compatibility"
the customer will still be able to use both keys (the old and the new), while the application will use a single entry (the new key), and both of them will "point" to the same value.

Reply:
assuming:
* `a` is an alias of `b`
* looking for the value of `b` -> will return the value of `b`
* looking for the the value of `a` -> will change `a` to `b` -> will return the value of `b`

the problem is the opposite
the customer can set `a` or `b` in the configuration
we want to look for the value of `a`, and to get the value of `a` if exists, otherwise the value of `b`